app/src/main/java/com/jnj/vaccinetracker/common/util/FileUtil.kt

```diff
@@ -31,35 +34,82 @@ class FileUtil {
+                            context.packageName,
+                            uri,
+                            Intent.FLAG_GRANT_READ_URI_PERMISSION
+                        )
+                        // Use the original MediaStore URI
+                        openFile(context, uri, mimeType)
+                    } else {
+                        Toast.makeText(context, R.string.saved_file_but_not_found, Toast.LENGTH_LONG)
+                            .show()
                     }
-                    Toast.makeText(context, context.getString(R.string.visits_overview_saving_file_success_message), Toast.LENGTH_LONG).show()
-                    openFile(context, uri, mimeType)
                 } catch (e: Exception) {
-                    Log.e("ExportFile", "Error saving file", e)
-                    Toast.makeText(context, context.getString(R.string.visits_overview_saving_file_failure_message), Toast.LENGTH_LONG).show()
+                    Toast.makeText(
+                        context,
+                        context.getString(R.string.visits_overview_saving_file_failure_message),
+                        Toast.LENGTH_LONG
+                    ).show()
                 }
             } else {
-                Toast.makeText(context, context.getString(R.string.visits_overview_saving_file_failure_message), Toast.LENGTH_LONG).show()
+                Toast.makeText(
+                    context,
+                    context.getString(R.string.visits_overview_saving_file_failure_message),
+                    Toast.LENGTH_LONG
+                ).show()
             }
         }
 
-        fun openFile(context: Context, uri: Uri, mimeType: String) {
+        private fun openFile(context: Context, uri: Uri, mimeType: String) {
             try {
-                val intent = Intent(Intent.ACTION_VIEW).apply {
+                var intent = Intent(Intent.ACTION_VIEW).apply {
                     setDataAndType(uri, mimeType)
+                    setPackage("com.microsoft.office.excel")
                     addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+                    addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                }
+
+                if (intent.resolveActivity(context.packageManager) == null) {
+                    intent = Intent(Intent.ACTION_VIEW).apply {
+                        setDataAndType(uri, mimeType)
+                        setPackage("com.google.android.apps.docs")
+                        addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+                        addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+                    }
                 }
 
                 if (intent.resolveActivity(context.packageManager) != null) {
                     context.startActivity(intent)
                 } else {
-                    Toast.makeText(context, context.getString(R.string.visits_overview_no_available_app_message), Toast.LENGTH_LONG).show()
+                    val chooser = Intent.createChooser(
```

Comment:
did you test it this chooser option?

Reply:
Removed this option.

Reply:
I just asked :)

Reply:
ok, you probably mixed up comments, you removed docs option, not chooser option

Reply:
Yes i removed docs, and for chooser , we will need it in future.